### PR TITLE
Allow early stopping for autogluon.timeseries

### PR DESCRIPTION
**Pull request by huibinshen**
*Issue #, if available:*

This is continuation of https://github.com/awslabs/autogluon/pull/1795.
Make early stopping possible for autogluon.timeseries. By default, it is not enabled.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1917-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1917/2/index.html

File: timeseries/src/autogluon/timeseries/learner.py
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/learner.py:42]: Let's please fold `early_stopping_patience` into `hyperparameters`. It should not be an explicit constructor / fit argument for learners or models.

File: timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py:100]: see comment on learners.

File: timeseries/src/autogluon/timeseries/models/gluonts/callback.py
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/models/gluonts/callback.py:95]: is adaptive ES serializable? can we add a test for saving/loading models with AdaptiveES set?

File: timeseries/src/autogluon/timeseries/predictor.py
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/predictor.py:109]: similar to comment on learners, this should be provided during `fit` time and within `hyperparameters`.

File: timeseries/src/autogluon/timeseries/predictor.py
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/predictor.py:128]: predictors/learners should not have this as an instance variable as this can change from one `fit` call to another.

**Review by canerturkmen — COMMENTED** [review]
Thank you for the PR @huibinshen ! I left some comments, most of them to fold `early_stopping_patience` into hyperparameters.

File: timeseries/src/autogluon/timeseries/learner.py
Comment (huibinshen) [reply timeseries/src/autogluon/timeseries/learner.py:42]: Will change it.

File: timeseries/src/autogluon/timeseries/predictor.py
Comment (huibinshen) [reply timeseries/src/autogluon/timeseries/predictor.py:109]: Will change it in the next revision.

File: timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py
Comment (huibinshen) [reply timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py]: Changed.

File: timeseries/src/autogluon/timeseries/predictor.py
Comment (huibinshen) [reply timeseries/src/autogluon/timeseries/predictor.py]: Changed.

**Comment by szha** [comment]
Job PR-1917-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1917/3/index.html

File: timeseries/tests/unittests/models/test_es_callback.py
Comment (huibinshen) [reply timeseries/tests/unittests/models/test_es_callback.py]: That's a good point Caner, done.

**Comment by canerturkmen** [comment]
LGTM. I'll wait for CI before merging. Thanks!

**Review by canerturkmen — APPROVED** [review]

**Pull request merged**
